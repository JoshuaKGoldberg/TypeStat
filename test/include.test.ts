import path from "node:path";
import { describe, expect, it } from "vitest";

import { runMutationTest } from "../src/tests/testSetup.js";

describe("include", () => {
	it("asterisk", async () => {
		const caseDir = path.join(import.meta.dirname, "cases/include/asterisk");
		const { actualContent, expectedFilePath } = await runMutationTest(caseDir);
		await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
	});

	it("directory", async () => {
		const caseDir = path.join(import.meta.dirname, "cases/include/directory");
		const { actualContent, expectedFilePath } = await runMutationTest(caseDir);
		await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
	});
});
