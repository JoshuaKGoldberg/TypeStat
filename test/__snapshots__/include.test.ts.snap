// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`include > asterisk > options 1`] = `
"[
  {
    "cleanups": {
      "suppressTypeErrors": true
    },
    "files": {
      "above": "",
      "below": "",
      "renameExtensions": false
    },
    "filters": [],
    "fixes": {
      "importExtensions": false,
      "incompleteTypes": false,
      "missingProperties": false,
      "noImplicitAny": false,
      "noImplicitThis": false,
      "noInferableTypes": false,
      "strictNonNullAssertions": false
    },
    "hints": {
      "react": {
        "propTypes": "whenRequired",
        "propTypesOptionality": "asWritten"
      }
    },
    "include": [
      "<rootDir>/actual.ts",
      "<rootDir>/expected.ts",
      "<rootDir>/original.ts"
    ],
    "mutators": [
      [
        "./sampleMutator.cjs",
        null
      ]
    ],
    "output": {},
    "package": {
      "directory": "<rootDir>",
      "file": "<rootDir>/package.json"
    },
    "parsedTsConfig": {
      "options": {
        "configFilePath": "<rootDir>/tsconfig.json",
        "noEmit": true,
        "noImplicitAny": false,
        "noImplicitThis": false,
        "strictNullChecks": false
      },
      "fileNames": [
        "<rootDir>/actual.ts",
        "<rootDir>/expected.ts",
        "<rootDir>/original.ts"
      ],
      "typeAcquisition": {
        "enable": false,
        "include": [],
        "exclude": []
      },
      "raw": {
        "include": [
          "*"
        ]
      },
      "errors": [],
      "wildcardDirectories": {
        "<rootDir>": 0
      },
      "compileOnSave": false
    },
    "postProcess": {
      "shell": []
    },
    "projectPath": "<rootDir>/tsconfig.json",
    "types": {}
  }
]"
`;

exports[`include > directory > options 1`] = `
"[
  {
    "cleanups": {
      "suppressTypeErrors": true
    },
    "files": {
      "above": "",
      "below": "",
      "renameExtensions": false
    },
    "filters": [],
    "fixes": {
      "importExtensions": false,
      "incompleteTypes": false,
      "missingProperties": false,
      "noImplicitAny": false,
      "noImplicitThis": false,
      "noInferableTypes": false,
      "strictNonNullAssertions": false
    },
    "hints": {
      "react": {
        "propTypes": "whenRequired",
        "propTypesOptionality": "asWritten"
      }
    },
    "include": [
      "<rootDir>/actual.ts",
      "<rootDir>/expected.ts",
      "<rootDir>/original.ts"
    ],
    "mutators": [
      [
        "./sampleMutator.cjs",
        null
      ]
    ],
    "output": {},
    "package": {
      "directory": "<rootDir>",
      "file": "<rootDir>/package.json"
    },
    "parsedTsConfig": {
      "options": {
        "configFilePath": "<rootDir>/tsconfig.json",
        "noEmit": true,
        "noImplicitAny": false,
        "noImplicitThis": false,
        "strictNullChecks": false
      },
      "fileNames": [
        "<rootDir>/actual.ts",
        "<rootDir>/expected.ts",
        "<rootDir>/original.ts"
      ],
      "typeAcquisition": {
        "enable": false,
        "include": [],
        "exclude": []
      },
      "raw": {
        "include": [
          "."
        ]
      },
      "errors": [],
      "wildcardDirectories": {
        "<rootDir>": 1
      },
      "compileOnSave": false
    },
    "postProcess": {
      "shell": []
    },
    "projectPath": "<rootDir>/tsconfig.json",
    "types": {}
  }
]"
`;

exports[`include > sub-directory > options 1`] = `
"[
  {
    "cleanups": {
      "suppressTypeErrors": false
    },
    "files": {
      "above": "",
      "below": "",
      "renameExtensions": false
    },
    "filters": [],
    "fixes": {
      "importExtensions": false,
      "incompleteTypes": false,
      "missingProperties": false,
      "noImplicitAny": false,
      "noImplicitThis": false,
      "noInferableTypes": false,
      "strictNonNullAssertions": false
    },
    "hints": {
      "react": {
        "propTypes": "whenRequired",
        "propTypesOptionality": "asWritten"
      }
    },
    "include": [
      "<rootDir>/code/import.ts",
      "<rootDir>/actual.ts",
      "<rootDir>/original.ts"
    ],
    "mutators": [
      [
        "./sampleMutator.cjs",
        null
      ]
    ],
    "output": {},
    "package": {
      "directory": "<rootDir>",
      "file": "<rootDir>/package.json"
    },
    "parsedTsConfig": {
      "options": {
        "configFilePath": "<rootDir>/tsconfig.json",
        "noEmit": true,
        "noImplicitAny": false,
        "noImplicitThis": false,
        "strictNullChecks": false
      },
      "fileNames": [
        "<rootDir>/code/import.ts",
        "<rootDir>/actual.ts",
        "<rootDir>/original.ts"
      ],
      "typeAcquisition": {
        "enable": false,
        "include": [],
        "exclude": []
      },
      "raw": {
        "include": [
          "code",
          "actual.ts",
          "original.ts"
        ]
      },
      "errors": [],
      "wildcardDirectories": {
        "<rootDir>/code": 1
      },
      "compileOnSave": false
    },
    "postProcess": {
      "shell": []
    },
    "projectPath": "<rootDir>/tsconfig.json",
    "types": {}
  },
  {
    "cleanups": {
      "suppressTypeErrors": true
    },
    "files": {
      "above": "",
      "below": "",
      "renameExtensions": false
    },
    "filters": [],
    "fixes": {
      "importExtensions": false,
      "incompleteTypes": false,
      "missingProperties": false,
      "noImplicitAny": false,
      "noImplicitThis": false,
      "noInferableTypes": false,
      "strictNonNullAssertions": false
    },
    "hints": {
      "react": {
        "propTypes": "whenRequired",
        "propTypesOptionality": "asWritten"
      }
    },
    "include": [
      "<rootDir>/code/import.ts",
      "<rootDir>/actual.ts",
      "<rootDir>/original.ts"
    ],
    "mutators": [
      [
        "fixImportExtensions",
        null
      ],
      [
        "fixIncompleteTypes",
        null
      ],
      [
        "fixMissingProperties",
        null
      ],
      [
        "fixNoImplicitAny",
        null
      ],
      [
        "fixNoImplicitThis",
        null
      ],
      [
        "fixNoInferableTypes",
        null
      ],
      [
        "fixStrictNonNullAssertions",
        null
      ]
    ],
    "output": {},
    "package": {
      "directory": "<rootDir>",
      "file": "<rootDir>/package.json"
    },
    "parsedTsConfig": {
      "options": {
        "configFilePath": "<rootDir>/tsconfig.json",
        "noEmit": true,
        "noImplicitAny": false,
        "noImplicitThis": false,
        "strictNullChecks": false
      },
      "fileNames": [
        "<rootDir>/code/import.ts",
        "<rootDir>/actual.ts",
        "<rootDir>/original.ts"
      ],
      "typeAcquisition": {
        "enable": false,
        "include": [],
        "exclude": []
      },
      "raw": {
        "include": [
          "code",
          "actual.ts",
          "original.ts"
        ]
      },
      "errors": [],
      "wildcardDirectories": {
        "<rootDir>/code": 1
      },
      "compileOnSave": false
    },
    "postProcess": {
      "shell": []
    },
    "projectPath": "<rootDir>/tsconfig.json",
    "types": {}
  }
]"
`;
